library/core/src/array/mod.rs

```diff
@@ -777,17 +821,70 @@ impl<T, const N: usize> [T; N] {
     /// right[1] = 4;
     /// assert_eq!(v, [1, 2, 3, 4, 5, 6]);
     /// ```
-    #[unstable(
-        feature = "split_array",
-        reason = "return type should have array as 2nd element",
-        issue = "90091"
-    )]
+    #[unstable(feature = "split_array", reason = "new API", issue = "90091")]
     #[inline]
-    pub fn rsplit_array_mut<const M: usize>(&mut self) -> (&mut [T], &mut [T; M]) {
-        (&mut self[..]).rsplit_array_mut::<M>()
+    pub const fn rsplit_array_mut<const M: usize>(&mut self) -> (&mut [T; N - M], &mut [T; M]) {
+        // SAFETY: 0 <= (N-M) <= len (N)
+        let (left, right) = unsafe { self.split_at_mut_unchecked(N - M) };
+
+        // SAFETY: `split_at_mut_unchecked()` guarantees that:
+        // - `left` is a slice of `N-M` elements,
+        // - `right` is a slice of `N - (N-M) == M` elements.
+        unsafe { (from_mut_slice_unchecked(left), from_mut_slice_unchecked(right)) }
     }
 }
 
+/// Unsafely converts a slice of `N` elements to an array reference of `N` elements.
+///
+/// # Safety
+///
+/// The caller must ensure that the slice has exactly `N` elements. Violating this constraint
```

Comment:
I think this should say "the slice has at least `N` elements". Calling it is sound if it has more.